src/components/questions/FormAnswer.jsx

```diff
@@ -0,0 +1,84 @@
+import React, {useState, useEffect, useRef} from 'react';
+import styles from './questions.module.css';
+import {useProductContext} from '../../context/ProductContext.jsx';
+
+const AnswerForm = (props) => {
+
+  // Store variables from context
+  const {info} = useProductContext();
+  const productName = info.name;
+
+  // This section keeps track of the uploaded file images in an array. I use this state to prevent more than 5 files from being uploaded.
+  const [uploadArray, setUploadArray] = useState([]);
+  // let imageInput = document.getElementById('image-uploads');
+
+  // useEffect(() => {
+  // }, [uploadArray])
+
+  const handleUpload = (e) => {
+    console.log(e.target.files[0])
+    setUploadArray(uploadArray => [...uploadArray, e.target.files[0]]);
+    uploadFile(e);
+    // updateArray(uploadArray => [...uploadArray, e.target.files[0]])
+  }
+  // These next functions allow me to track the latest file uploaded and store it in the 'file' variable. I use this to turn the file into an image at the bottom of the form where you see an image tag.
+  const [file, setFile] = useState()
+
+  const uploadFile = (e) => {
+    setFile(e.target.files[0])
+  }
+  // Create state for input fields -- will use this for validation later
+  const [answer, updateAnswer] = useState('');
+  const handleAnswerChange = (e) => {
+    updateAnswer(e.target.value);
+  }
+  const [nickname, updateNickname] = useState('');
+  const handleNicknameChange = (e) => {
+    updateNickname(e.target.value);
+  }
+  const [email, updateEmail] = useState('');
+  const handleEmailChange = (e) => {
+    updateEmail(e.target.value);
+  }
+
+  const handleSubmit = () => {
+    alert(`answer: ${answer} nickname: ${nickname} email ${email}`)
+  }
+
+
+  if (props.show) {
```

Comment:
this conditional statement you can put it to AddAnswer.jsx line 16, to determine render this component or not, by doing so we may save a bit of memory and dry the code. You can get rid of the line 79~81 in this file as well. 

// if modalShown is truthy,  render <AnserForm>, else nothing ( <></>)
// https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/basic-javascript/use-the-conditional-ternary-operator

{modalShown?  <AnswerForm  modalButtonClick={modalButtonClick}/> : <></>}

Reply:
{modalShown && <AnswerForm  modalButtonClick={modalButtonClick}/>}  will make it cleaner than ever! Inspired by your pervious code